app/templates/scripts/application.erb

```diff
@@ -26,7 +26,7 @@ after_bundle do
   if yes?('Test + start application?')
     run 'bin/rspec'
     run 'yarn test'
```

Comment:
When I created a new app and said `yes` to `Test + start application?`

```
tryan@tryan-mbp rails-7-app % yarn test
yarn run v1.22.17
warning package.json: No license field
error Command "test" not found. Did you mean "jest"?
```
Is this expected?

Reply:
I did not run into that ... when I said `y` to that, it ran rspec and jest and kicked it off. I'll do it again right now and see if I can see how it might not work.

Reply:
it makes me feel like the `app/templates/files/inserts/package_scripts.json` directives didn't get inserted into the `package.json` file under the `scripts` section. That's why it would not understand `yarn test` from within the app.

Yeah, this is what it looks like for me at this point:
```
Test + start application? y
         run  bin/rspec from "."
Run options: exclude {:wcag=>true}
.....

Finished in 4.07 seconds (files took 13.82 seconds to load)
5 examples, 0 failures

Coverage report generated for RSpec to /Users/JGarguilo/work/rails-7-app/coverage. 9 / 9 LOC (100.0%) covered.
         run  yarn test from "."
yarn run v1.22.17
warning package.json: No license field
$ jest --config package.json --passWithNoTests
No tests found, exiting with code 0
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |
----------|---------|----------|---------|---------|-------------------
✨  Done in 5.02s.
         run  bin/dev from "."
14:56:38 web.1  | started with pid 61155
14:56:38 js.1   | started with pid 61156
14:56:38 js.1   | yarn run v1.22.17
14:56:38 js.1   | warning package.json: No license field
14:56:38 js.1   | $ webpack --config webpack.config.js --watch
...
```

Hm, hm, hm. Alright, I nuked that and am rerunning it again, I want to try something else at that step. Then I'll come back with some other possible stuff from within the generated app. Wondering if there's a system prerequisite I missed.

Reply:
Still worked ... I wonder if something happened where the `build` job didn't get added to `package.json`, it should have something that looks like this:

```
  "scripts": {
    "build": "webpack --config webpack.config.js"
,
    "css-fix": "stylelint --fix app/**/*.*css",
    "test": "jest --config package.json --passWithNoTests"
  }  ,
```

The `build` job is generated by rails, the other two jobs get added by our generator ... hmmmmmm ... what would do that ... the `-j webpack` bit of the `rails new` should be putting the `package.json` together correctly.

Ah, ah, ah ... I got the failure, but I got it by running `rails _6.1.4.6_ new rails-7-app -T -O -j webpack --skip-turbolinks -M --skip-action-cable --template "http://localhost:3000/templates/eyJjaSI6ZmFsc2UsImRiIjpmYWxzZSwidWkiOnRydWUsIm1haWxlciI6ZmFsc2UsImFjdGlvbl9jYWJsZSI6ZmFsc2UsImFwaSI6ZmFsc2UsImRpc3BsYXlOYW1lIjoiUmFpbHMgNyBBcHAiLCJ0eXBlIjoiYXBwbGljYXRpb24ifQ=="`

It wound up creating a Rails 7 app according to the Gemfile, but by me telling it to use Rails 6.1.4.6 it didn't do the expected installation of the webpack stuff.

If you do `rails -v` do you get `7.0.2.2`, or does it tell you like `6.1.4.6` ... maybe it's trying to run with Rails 6 instead of Rails 7?

Reply:
I get Rails 7.0.2.2.

I just tried again and ran into the same error. Maybe its just me?

Reply:
That's confusing, definitely. I was able to get the error, but only by running the rails new through my `6.1.4.6` version.

But Rails 7 with the javascript flag should handle it just fine. Well, even less the templating script maybe there's something to discover here.

The `test` script only knows where to inject into the `package.json` if a `build` script exists first. So, if the `rails new` isn't building that, then we can never hook in our custom stuff.

So, if I run `rails new rails-7-app -T -O -j webpack --skip-turbolinks -M --skip-action-cable`, so basically just dropping the template script ... I see that I get ...

yeah, an app that has a `package.json` that looks like this:
```
{
  "name": "app",
  "private": "true",
  "dependencies": {
    "@hotwired/stimulus": "^3.0.1",
    "@hotwired/turbo-rails": "^7.1.1",
    "webpack": "^5.70.0",
    "webpack-cli": "^4.9.2"
  },
  "scripts": {
    "build": "webpack --config webpack.config.js"
  }
}
```

Which is a lot less than what we modify it to be, but the critical thing is that the `build` part is there in the `scripts` key. At least, for this part ...

If I run `rails _6.1.4.6_ new rails-7-app -T -O -j webpack --skip-turbolinks -M --skip-action-cable`, then I get a package.json like ...

```
{
  "name": "rails-7-app",
  "private": true,
  "dependencies": {
    "@rails/ujs": "^6.0.0",
    "@rails/webpacker": "5.4.3",
    "webpack": "^4.46.0",
    "webpack-cli": "^3.3.12"
  },
  "version": "0.1.0",
  "devDependencies": {
    "webpack-dev-server": "^3"
  }
}
```

So no `build` to hook into there ... but maybe there's something else failing. I guess that's one thing to check, does your app have a `build` in `scripts` in package.json ... if so, then something else may have failed along the way. I'm not sure what that could be. A node or yarn issue or something? Although, actually when I uses 6.1.x, the output celebrates "Webpacker successfully installed 🎉 🍰" ... which is def not what we want. If "webpacker" appears anywhere in your app source, then it's Rails 6ing you despite the version report.

If the vanilla `rails new` without the template stuff at least got the bare `package.json` with the `build` script and no webpacker, we could look at executing through the steps of the script to see where the breakage might be.

If the vanilla rails new doesn't work ... well, I guess we could try to force a version and do a `rails _7.0.2.2_ new ...` ... I'll try that out to see what it does for me ...

Yep, I get a working app. V v v confusing. Well, I guess there might be something I'm missing. If all the `package.json` stuff looks good, the `btoa` encryption _should_ be the same between our computers. You could toss me your rails new and I could see if I missed something/get it to break the same.